couchbase-entity/src/main/java/com/kaufland/Logger.kt

```diff
@@ -41,4 +41,15 @@ class Logger internal constructor(processingEnv: ProcessingEnvironment) {
     fun hasErrors(): Boolean {
         return hasErrors
     }
+
+    fun abortWithError(e: PostValidationException) {
+        if(e.causingElements.isNotEmpty()){
+            for (causingElement in e.causingElements) {
+                error(e.message ?: "unknown", causingElement)
+            }
+        }else{
+         error(e.message ?: "unknown", null)
+        }
```

Comment:
Would it not make sense, to always log the exception itself; even in addition to the causingElements?
It can still have its own message, if I understood the code correctly

Reply:
In most of the cases the "User" is not interested in the exception trace itself therefore I decided to only print the error message and the causing element.
For interested User it's always possible to see the stack if they compile with --stacktrace since the exception keeps unhandled in the end.